core/src/main/java/org/infinispan/factories/threads/CoreExecutorFactory.java

```diff
@@ -0,0 +1,26 @@
+package org.infinispan.factories.threads;
+
+import java.util.concurrent.ExecutorService;
+
+import org.infinispan.commons.executors.BlockingThreadPoolExecutorFactory;
+import org.infinispan.commons.executors.ThreadPoolExecutorFactory;
+
+public class CoreExecutorFactory {
+   private CoreExecutorFactory() { }
+
+   public static ThreadPoolExecutorFactory<? extends ExecutorService> executorFactory(int maxThreads, int coreThreads,
+         int queueLength, long keepAlive, boolean nonBlocking) {
+      if (nonBlocking) {
+         return new BlockingThreadPoolExecutorFactory(maxThreads, coreThreads, queueLength, keepAlive, nonBlocking);
```

Comment:
I suggest renaming BlockingThreadPoolExecutorFactory, using it only on the nonBlocking path looks weird.

Reply:
Yeah, I had debated about renaming it to `NonBlockingThreadPoolExecutorFactory`. I will go ahead and do that and remove the blocking arguments.

Reply:
Actually I can't rename it as it is a public package. I am thinking instead I will just deprecate the one in commons and add the new class in core where it is private. We can move it later if needed.